extensions/libraries/redcore/api/api.php

```diff
@@ -199,9 +199,27 @@ public static function getPostedData()
 				}
 			}
 
-			// We try to transform it into XML
-			if (is_null($parsedData) && $xml = @simplexml_load_string($inputData))
+			$xml = @simplexml_load_string($inputData);
+
+			// We try to transform soap XML
+			if (is_null($parsedData) && $xml !== false)
 			{
+				/*
+					XML Namespaces suck and I couldn't get it to read the data accurately for SOAP
+					So I'm stripping everything between < and : in the xml and re-rendering it
+				    I know. I know. My skin is crawling too, but its the best I could do at the moment.
+				    If someone can make this work with namespaces that would be B-E-A-UTIFUL!
+				*/
+				$cleanData = preg_replace('/<(\/)?([a-z0-9]+):([a-z0-9]+)/i', '<$1$3', $inputData);
+				$xml = @simplexml_load_string($cleanData);
```

Comment:
Thanks for the input @Kixo 
Unfortunately ```$xml = @simplexml_load_string($inputData)``` doesn't work for xml that is namespaced. After scratching my head for awhile I went to the PHP manual to figure out why this method was being skipped. That is when I noticed the warning that says 

    "This function may return Boolean FALSE, but may also return a non-Boolean value which evaluates to FALSE."  

What was happening is that SOAP requests were not being processed at all and instead the API $data var was being set as just the raw xml. However by using ```$xml !== false``` then the SOAP input is processed as XML.

As I said in the comments, I don't like loading the xml twice, but I'm not familiar enough with xml namespaces to come up with a solution that would work for any possible namespaced xml sent. 

So once we've confirmed that it is actually xml data we are being sent, I just remove the namespace with the preg_replace from the string and reload the xml. Since we weren't using the namespace data for anything. It seemed like an acceptable compromise. But if you have a cleaner solution, I would defiantly welcome it. 

I'm 98% sure this conditional won't evaluate to true for JSON data, but since I have to get it to pass the tests anyway, I'll make sure I check that concern tonight.